pkg/networkmanager/interfaces.js

```diff
@@ -421,63 +443,52 @@
     /* NetworkManager specific data conversions and utility functions.
      */
 
-    function ip4_address_from_nm(addr) {
-        return [utils.ip4_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip4_to_text(addr[2], true)
-        ];
+    function ip_address_from_nm(addr) {
+        return {
+            address: addr.address.v,
+            prefix: utils.ip_prefix_to_text(addr.prefix.v)
+        };
     }
 
-    function ip4_address_to_nm(addr) {
-        return [utils.ip4_from_text(addr[0]),
-            utils.ip4_prefix_from_text(addr[1]),
-            utils.ip4_from_text(addr[2], true)
-        ];
-    }
+    function ip_address_to_nm(addr, ipv) {
+        const prefix = ipv === "ipv4" ? utils.ip4_prefix_from_text(addr.prefix) : utils.ip_prefix_from_text(addr.prefix);
 
-    function ip4_route_from_nm(addr) {
-        return [utils.ip4_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip4_to_text(addr[2], true),
-            utils.ip_metric_to_text(addr[3])
-        ];
-    }
+        if (!utils.validate_ip(addr.address)) {
+            throw cockpit.format(_("Invalid IP address: $0"), addr.address);
+        }
 
-    function ip4_route_to_nm(addr) {
-        return [utils.ip4_from_text(addr[0]),
-            utils.ip4_prefix_from_text(addr[1]),
-            utils.ip4_from_text(addr[2], true),
-            utils.ip_metric_from_text(addr[3])
-        ];
-    }
-    function ip6_address_from_nm(addr) {
-        return [utils.ip6_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip6_to_text(addr[2], true)
-        ];
+        return {
+            address: { t: "s", v: addr.address },
+            prefix: { t: "u", v: prefix },
+        };
     }
 
-    function ip6_address_to_nm(addr) {
-        return [utils.ip6_from_text(addr[0]),
-            parseInt(addr[1], 10) || 64,
-            utils.ip6_from_text(addr[2], true)
-        ];
+    function route_from_nm(route) {
+        return {
+            dest: route.dest.v,
+            prefix: utils.ip_prefix_to_text(route.prefix.v),
+            next_hop: route["next-hop"].v,
+            metric: utils.ip_metric_to_text(route.metric.v),
+        };
     }
 
-    function ip6_route_from_nm(addr) {
-        return [utils.ip6_to_text(addr[0]),
-            utils.ip_prefix_to_text(addr[1]),
-            utils.ip6_to_text(addr[2], true),
-            utils.ip_metric_to_text(addr[3]),
-        ];
-    }
+    function route_to_nm(route, ipv) {
+        const prefix = ipv === "ipv4" ? utils.ip4_prefix_from_text(route.prefix) : utils.ip_prefix_from_text(route.prefix);
+
+        if (!utils.validate_ip(route.dest)) {
+            throw cockpit.format(_("Invalid destination address: $0"), route.dest);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-1fe4fb60-20250903-092011-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)